File: dts-playlist-inf/src/main/java/com/gugu/dts/playlist/inf/entity/Song.java
package com.gugu.dts.playlist.inf.entity;

import lombok.Data;

/**
 * Table: song
 */
@Data
public class Song {
    /**
     * Column: id
     */
    private Integer id;

    /**
     * Column: name
     */
    private String name;

    /**
     * Column: path
     */
    private String path;

    /**
     * Column: bpm
     */
    private Double bpm;

    /**
     * Column: used_times
     */
    private Integer usedTimes;

    /**
     * Column: library_id
     */
    private Integer libraryId;

    /**
     * Column: track_length
     */
    private Integer trackLength;

    /**
     * Column: artist
     */
    private String artist;

    /**
     * Column: create_at
     */
    private String createAt;

    /**
     * Column: update_at
     */
    private String updateAt;
}

File: dts-playlist-core/src/main/kotlin/com/gugu/dts/playlist/core/entity/Song.kt
package com.gugu.dts.playlist.core.entity

import com.gugu.dts.playlist.api.`object`.ISong

data class Song(
        override val name: String,
        override val path: String,
        override val usedTimes: Int = 0,
        override val bpm: Double? = null,
        override val artist: String? = null,
        override val trackLength: Int = 0,
        override val id: Int? = null
) : ISong